auth/domain/src/test/java/com/pronoidsoftware/auth/domain/UserDataValidatorTest.kt

```diff
@@ -0,0 +1,90 @@
+    private lateinit var emailPatternValidatorFake: PatternValidator
+
+    @BeforeEach
+    fun setUp() {
+        // we don't need to check that the email pattern matcher works, so returning true
+        emailPatternValidatorFake = object : PatternValidator {
+            override fun matches(value: String): Boolean {
+                return true
+            }
+        }
+        sut = UserDataValidator(emailPatternValidatorFake)
+    }
+
+    @Test
+    fun `validateName returns true when name length is within valid range`() {
+        val result = sut.validateName("ValidName")
+        assertThat(result).isTrue()
+    }
+
+    @Test
+    fun `validateName returns true when name length is within valid range with whitespace`() {
+        val result = sut.validateName("Valid Name ")
+        assertThat(result).isTrue()
+    }
+
+    @Test
+    fun `validateName returns false when name length is less than minimum`() {
+        val result = sut.validateName("abc")
+        assertThat(result).isFalse()
+    }
+
+    @Test
+    fun `validateName returns false when name length is more than maximum`() {
+        val result = sut.validateName("a".repeat(51))
+        assertThat(result).isFalse()
+    }
+
+    @Test
+    fun `validateName returns false when name length is less than minimum with whitespace`() {
+        val result = sut.validateName("  ab ")
+        assertThat(result).isFalse()
+    }
+
+    @Test
+    fun `validatePassword returns valid state when password meets all requirements`() {
+        val result = sut.validatePassword("ValidPassword1")
+        assertThat(result.hasMinimumLength).isTrue()
+        assertThat(result.hasDigit).isTrue()
+        assertThat(result.hasLowerCaseCharacter).isTrue()
+        assertThat(result.hasUpperCaseCharacter).isTrue()
+    }
+
+    @Test
+    fun `validatePassword returns valid when password meets all requirements with whitespace`() {
+        val result = sut.validatePassword("Valid Password 1")
+        assertThat(result.hasMinimumLength).isTrue()
+        assertThat(result.hasDigit).isTrue()
+        assertThat(result.hasLowerCaseCharacter).isTrue()
+        assertThat(result.hasUpperCaseCharacter).isTrue()
+    }
+
+    @Test
+    fun `validatePassword returns valid when password meets all requirements with specials`() {
+        val result = sut.validatePassword("Valid@Password_1")
+        assertThat(result.hasMinimumLength).isTrue()
+        assertThat(result.hasDigit).isTrue()
+        assertThat(result.hasLowerCaseCharacter).isTrue()
+        assertThat(result.hasUpperCaseCharacter).isTrue()
+    }
+
+    @Test
+    fun `validatePassword returns invalid state when password does not meet requirements`() {
+        val result = sut.validatePassword("invalid")
+        assertThat(result.hasMinimumLength).isFalse()
+        assertThat(result.hasDigit).isFalse()
+        assertThat(result.hasLowerCaseCharacter).isTrue()
+        assertThat(result.hasUpperCaseCharacter).isFalse()
+    }
```

Comment:
With JUnit5 you can do parameterized tests which let you specify a list of inputs with expected outputs, so you could just have a single test that tests everything of the above 4 tests here (and more if you want). I love this for tests where the inputs and outputs can be easily specified.

Something like this should work:
```
@ParameterizedTest
@CsvSource(
    "ValidPassword1, true",
    "Valid Password 1, true",
    "Valid@Password_1, true",
    "invalid, false",
)
fun testValidatePassword(password: String, expectedIsValid: Boolean) {

}
```
I'd also put some more focus on testing edge cases, like empty strings, short strings with special chars and numbers, only uppercase, only lowercase, etc. There are many more cases where something could go wrong. Your tests for example wouldn't catch if the real function would let a password like `Invalid1` pass.